build/repo.targets

```diff
@@ -1,56 +1,195 @@
+
+    <_BuildGraphFile>$(BuildDir)BuildGraph.proj</_BuildGraphFile>
+    <_CloneRepositoryRoot>$(RepositoryRoot).r\</_CloneRepositoryRoot>
+    <_DependencyPackagesDirectory>$(RepositoryRoot).deps\build\</_DependencyPackagesDirectory>
+    <_CommitsFile>$(RepositoryRoot).deps\build\RepositoryCommits.proj</_CommitsFile>
+    <_RestoreGraphSpecsDirectory>$(RepositoryRoot)obj\package-specs\</_RestoreGraphSpecsDirectory>
+
+    <_RepositoryBuildTargets Condition="'$(_RepositoryBuildTargets)'=='' AND '$(CompileOnly)'=='true'">/t:Package /t:VerifyPackages</_RepositoryBuildTargets>
+    <_RepositoryBuildTargets Condition="'$(_RepositoryBuildTargets)'==''">/t:Verify</_RepositoryBuildTargets>
+    <_RepositoriesWithCommitsFile>$(RepositoryRoot).deps\build\Repositories.props</_RepositoriesWithCommitsFile>
+
+    <PrepareDependsOn>$(PrepareDependsOn);FilterRepositories</PrepareDependsOn>
+    <CleanDependsOn>$(CleanDependsOn);CleanUniverseArtifacts</CleanDependsOn>
+    <BuildDependsOn>$(BuildDependsOn);CloneRepositories;BuildRepositories</BuildDependsOn>
   </PropertyGroup>
 
-  <!-- workaround. Using Sake as the intermediate means this property doesn't flow into repo builds. -->
-  <Target Name="SetBuildNumber" Condition="'$(BuildNumber)' != ''">
-    <SetEnvironmentVariable Variable="BuildNumber" Value="$(BuildNumber)" />
+  <Import
+    Project="$(_RepositoriesWithCommitsFile)"
+    Condition="Exists('$(_RepositoriesWithCommitsFile)')" />
+  <Import
+    Project="Repositories.props"
+    Condition="!Exists('$(_RepositoriesWithCommitsFile)')" />
+  <Import
+    Project="$(_CommitsFile)"
+    Condition="Exists('$(_CommitsFile)')" />
+
+  <Target Name="CleanUniverseArtifacts">
+    <RemoveDir Directories="obj" Condition="Exists('obj')" />
+    <RemoveDir Directories="$(_CloneRepositoryRoot)" Condition="Exists('$(_CloneRepositoryRoot)')" />
+    <RemoveDir Directories="$(_ArtifactsPackagesDirectory)" Condition="Exists('$(_ArtifactsPackagesDirectory)')" />
   </Target>
 
-  <Target Name="VerifyAll">
-    <PropertyGroup>
-      <_SakeTargets>$(_SakeTargets):verify-all</_SakeTargets>
-    </PropertyGroup>
-    <MSBuild Projects="$(MSBuildProjectFullPath)"
-      Targets="Sake"
-      Properties="SakeTargets=$(_SakeTargets);RepositoryRoot=$(RepositoryRoot)" />
+  <Target Name="FilterRepositories">
+    <ItemGroup Condition="'$(KOREBUILD_REPOSITORY_INCLUDE)'!=''">
+      <_RepositoriesToInclude Include="$(KOREBUILD_REPOSITORY_INCLUDE)" />
+      <Repository
+        Remove="@(Repository)"
+        Condition="'@(Repository)'!='@(_RepositoriesToInclude)' AND '%(Identity)'!=''" />
+    </ItemGroup>
+
+    <ItemGroup Condition="'$(KOREBUILD_REPOSITORY_EXCLUDE)'!=''">
+      <RepositoriesToExclude Include="$(KOREBUILD_REPOSITORY_EXCLUDE)" />
+      <Repository Remove="@(RepositoriesToExclude)" />
+    </ItemGroup>
+
+    <Warning Text="KOREBUILD_REPOSITORY_EXCLUDE AND KOREBUILD_REPOSITORY_INCLUDE are specified."
```

Comment:
I think you want to pick one or the other. It's likely a bad configuration when both are specified. I could make this an Error